src/dbnode/encoding/m3tsz/timestamp_iterator.go

```diff
@@ -133,11 +131,11 @@ func (it *TimestampIterator) readNextTimestamp(stream encoding.IStream) error {
 	}
 
 	it.PrevTimeDelta += dod
-	it.PrevTime = it.PrevTime.Add(it.PrevTimeDelta)
+	it.PrevTime = it.PrevTime + it.PrevTimeDelta
 	return nil
 }
 
-func (it *TimestampIterator) tryReadMarker(stream encoding.IStream) (time.Duration, bool, error) {
+func (it *TimestampIterator) tryReadMarker(stream encoding.IStream) (xtime.UnixNano, bool, error) {
```

Comment:
Similarly here can you just keep the return of this method `tryReadMarker ` as `time.Duration`?